frameposter.sh

```diff
@@ -114,48 +105,55 @@ nth(){
-	secs="${1##*:}" milisecs="${secs#*.}" secs="${secs%%.*}"
-	[[ -z "${list}" ]] && list="${subs_content}"
-	while read -r b; do
-		start="$(toepoch "$([[ "${b}" =~ ^([^\,]*), ]] && printf '%s' "${BASH_REMATCH[1]}")")"
-		end="$(toepoch "$([[ "${b}" =~ ,([^\€]*)€ ]] && printf '%s' "${BASH_REMATCH[1]}")")"
-		if (( $(bc -l <<< "${a} >= ${start}") )) && (( $(bc -l <<< "${a} <= ${end}") )); then
-			if [[ "${b}" =~ [^\,]*\,sign ]]; then
-				message_craft="【$(sed -nE 's_[[:blank:]]{3}_ _g;s_\!([a-zA-Z0-9])_\! \1_g;s_.*\(\)(.*)_\1_p' <<< "${b}")】
+	# Set the current time variable
+	current_time="${1}"
+	hrs="${current_time%%:*}"
+	mins="${current_time%:??.??}"
+	mins="${mins/*:/}"
+
+	# Convert the current time to seconds
+	IFS=':.' read -r h m s ms <<< "$current_time"
+	current_time_seconds="$((10#$h * 3600 + 10#$m * 60 + 10#$s + 10#$ms / 100)).$ms"
+
+	# Scrape the time of the subtitle
+	while IFS='×' read -r start_time end_time speaker subtitle; do
+	    IFS=':.' read -r h m s ms <<< "$start_time"
+	    start_time_seconds="$((10#$h * 3600 + 10#$m * 60 + 10#$s + 10#$ms / 100)).$ms"
+	    IFS=':.' read -r h m s ms <<< "$end_time"
+	    end_time_seconds="$((10#$h * 3600 + 10#$m * 60 + 10#$s + 10#$ms / 100)).$ms"
+	    # Check if the current time is between the start and end time
+	    if awk -v a="$current_time_seconds" -v b="$start_time_seconds" 'BEGIN{{if(a>=b) exit 0;exit 1}}' && awk -v a="$current_time_seconds" -v b="$end_time_seconds" 'BEGIN{{if(a<=b) exit 0;exit 1}}'; then
+	        # Strip the stylings and display the subtitle
+	        subtitle="${subtitle/\{*\}/}"
+	        subtitle="${subtitle/\\N/}"
+	        if [[ "${speaker}" =~ [^\,]*\,sign ]]; then
+	            message_craft="【$(sed -E 's_[[:blank:]]{3}_ _g;s_\!([a-zA-Z0-9])_\! \1_g' <<< "${subtitle}")】
 ${message_craft}"
-			elif [[ "${b}" =~ Signs\,\, ]]; then
-				message_craft="$(sed -nE 's_[[:blank:]]{3}_ _g;s_\!([a-zA-Z0-9])_\! \1_g;s_.*\(\)(.*)_"\1"_p' <<< "${b}")
+	        elif [[ "${speaker}" =~ Signs\,\, ]]; then
+	            message_craft="\"$(sed -E 's_[[:blank:]]{3}_ _g;s_\!([a-zA-Z0-9])_\! \1_g' <<< "${subtitle}")\"
 ${message_craft}"
-			elif [[ "${b}" =~ Songs_OP\,OP ]]; then
-				is_opsong="1"
-				message_craft="『$(sed -nE 's_[[:blank:]]{3}_ _g;s_\!([a-zA-Z0-9])_\! \1_g;s_.*\(\)(.*)_\1_p' <<< "${b}")』
+	        elif [[ "${speaker}" =~ Songs_OP\,OP ]]; then
+	            is_opsong="1"
+	            message_craft="『$(sed -E 's_[[:blank:]]{3}_ _g;s_\!([a-zA-Z0-9])_\! \1_g' <<< "${subtitle}")』
 ${message_craft}"
-			else
-				message_craft="${b/*\(\)/}
+	        else
+	            message_craft="${subtitle}
 ${message_craft}"
-			fi
-			continue
-		fi
-		(( $(bc -l <<< "${a} <= ${end}") )) && break
-	done <<-EOF
-	${list}
-	EOF
+	        fi
+	    fi
+	done < <(sed -nE "/Dialogue:.*0,${hrs}:${mins}:??.??/"' s_.*\,([^\,]*),([^\,]*),(.*,[^\,]*,)0000,0000,0000,,(.*)_\1×\2×\3×\4_p' subtitle.ass)
```

Comment:
Isn't `subtitle.ass` supposed to be `"${locationsub}"`

Reply:
> Isn't `subtitle.ass` supposed to be `"${locationsub}"`

ww